C-Basics-master/src/countFactorials.cpp

```diff
@@ -0,0 +1,44 @@
+/*Problem - To find the number of factorials in a given range.
+
+Description - You will be given an input array[l, h] which gives the range.Your job is to return the count of the number of factorials between 'l' and 'h' both included.
+
+Example: input = [2, 10]
+		 The factorails between 2 and 10 is '2' alone.Hence 1 must be returned.
+
+
+Constraints :
+1) The range should not be negative, i.e neither of l and h should be negative.If so, return -1.
+2) l, h are both long integers.
+3) The range should be non decreasing, i.e l <= h.It is not, return -1.
+
+DONT CHANGE THE FUNCTION HEADERS.l
+
+
+
+*/
+
+#include <stdio.h>
+
+
+int countFactorials(long l, long h)
+{
+	if (l < 0 && h < 0)
+		return -1;
+	int count;
+	int n = h;
+	int *ar = new int [n]; 
+	int f = 1;
+	for (int i = l; i <= h; i++)
+	{
+		f = f*i;
+		ar[i] = f;
+		if (ar[i] == i)
+			count++;
+		else
+			return -1;
```

Comment:
What is the necessity to store all factorials in an array? You are just using it in one comparision. Can't you do it without using an array? Because if the high number is too big then it's likely that you are occupying too much memory by initilaising such a huge array and not using it to the fullest.